### PR TITLE
Fixes #8 fixes #4 fixes #5 - Implementing .NET test suite for libskycoin-dotnet

### LINKED ISSUE #4: Generate Libskycoin.net source code using Swig
- Generate Libskycoin.net source code using Swig

### LINKED ISSUE #8: Implement libskycoin test suite for Libskycoin-dotnet
- [x] Tests for `cipher`
- [x] Tests for `coin`
- [x] Tests for `droplet`
- [x] Tests for `fee`

### PR DESCRIPTION
**Pull request by stdevHan**
Fixes #8
Fixes #5 
Fixes #4

Changes:
- [x] Tests for `cipher`.
- [x] Tests for `coin`
- [x] Tests for `fee`
- [x] Tests for `droplet`
- [x] Tests for `param`

### CODE REVIEW

**Comment by olemis** [comment]
Travis builds fail

**Comment by olemis** [comment]
Is this missing tests for `param` ?

File: swig/csharp_skycoin.mem.i [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+
+%rename(SKY_coin_Block_HashBody) CSharp_skycoin_SKY_coin_Block_HashBody;
+%inline {
+	GoUint32 CSharp_skycoin_SKY_coin_Block_HashBody(Block__Handle p0, cipher_SHA256* p1){
+		GoUint32 result = SKY_coin_Block_HashBody(p0,  p1);
+		return result;
+	}
+}
+
+// %rename(SKY_params_GetDistributionAddresses) CSharp_skycoin_SKY_params_GetDistributionAddresses;
```
Comment (olemis) [inline swig/csharp_skycoin.mem.i]: Why comments ?

File: swig/include/libskycoin.h [hunk 2]
```diff
@@ -0,0 +1,1233 @@
+/* Code generated by cmd/cgo; DO NOT EDIT. */
```
Comment (olemis) [inline swig/include/libskycoin.h]: Use the header file in `skycoin/skycoin` submodule .

Comment (olemis) [reply swig/include/libskycoin.h]: ... and remove this file .

File: swig/include/swig.h [hunk 3]
```diff
@@ -0,0 +1,56 @@
+
```
Comment (olemis) [inline swig/include/swig.h]: Use the definitions in `skycoin/skycoin` submodule , and delete this file .

File: swig/include/libskycoin.h (hunk 2 above)
Comment (stdevHan) [reply swig/include/libskycoin.h]: that I modify it, to use it with the lib, without error
https://github.com/simelo/libskycoin-dotnet/blob/stdevHan_t8_test_libdotnet/Makefile#L28

File: swig/csharp_skycoin.mem.i (hunk 1 above)
Comment (stdevHan) [reply swig/csharp_skycoin.mem.i]: I already deleted it

File: swig/include/swig.h (hunk 3 above)
Comment (stdevHan) [reply swig/include/swig.h]: I already deleted it

**Pull request merged**